src/Task.ts

```diff
@@ -78,6 +81,39 @@ export function handleTimeLockExpirationSet(event: TimeLockExpirationSet): void
   task.save()
 }
 
+export function handleCustomVolumeLimitSet(event: CustomVolumeLimitSet): void {
+  const task = Task.load(event.address.toHexString())
+  if (task == null) return log.warning('Missing task entity {}', [event.address.toHexString()])
+
+  const customVolumeLimitId = getVolumeLimitId(task, event.params.token)
+
+  const customVolumeLimit = new CustomVolumeLimit(customVolumeLimitId)
+  customVolumeLimit.task = task.id
+  customVolumeLimit.token = loadOrCreateERC20(event.params.token).id
+  customVolumeLimit.limitToken = loadOrCreateERC20(event.params.limitToken).id
+  customVolumeLimit.amount = event.params.amount
+  customVolumeLimit.period = event.params.period
+
+  customVolumeLimit.save()
+  task.save()
```

Comment:
Task saving is not needed

```suggestion
  customVolumeLimit.save()
```